content/docs/iac/concepts/secrets/write-only-fields.md

```diff
@@ -0,0 +1,76 @@
+---
+title_tag: "Write-Only Fields | Pulumi Concepts"
+meta_desc: This page provides an overview of write-only fields in Pulumi secrets handling.
+title: Write-only Fields
+h1: Write-only Fields
+meta_image: /images/docs/meta-images/docs-meta.png
+menu:
+    iac:
+        name: Write-Only Fields
+        parent: iac-concepts-secrets
+        weight: 1
+    concepts:
+        parent: secrets
+        weight: 1
+aliases:
+- /docs/intro/concepts/secrets/write-only-fields/
+- /docs/concepts/secrets/write-only-fields/
+---
+
+Write-only fields are resource properties that can be set during resource creation but are never returned by the cloud provider's API. This means Pulumi cannot read these values back from the cloud provider.
+
+## Write-only fields in Terraform
+
+Write-only fields are a concept that some providers inherit from their underlying Terraform providers. These providers have fields that are intentionally write-only for security reasons. For example, a database password might be set during creation but the provider will never return the actual password value in subsequent API calls.
+
+{{% notes type="info" %}}
+Pulumi supports these properties mainly for schema parity with the underlying Terraform provider. With Pulumi, sensitive data can always be stored in state as a Secret, which is the preferred approach when available.
+{{% /notes %}}
+
+## How Pulumi handles write-only fields
+
+When Pulumi encounters a write-only field:
+
+1. The value is used during resource creation or updates and sent to the cloud API.
+2. Its initial value gets written to Pulumi state inputs as a Secret. It will never appear in state outputs.
+3. On subsequent Read operations, the value will be set to `null`.
+4. On subsequent previews or updates, Pulumi will not detect or show diffs on these fields since they are not tracked in state.
+
+## Version control fields
+
+Some providers gate updates to write-only fields with a write-only _version_ field. This version field is under full Pulumi lifecycle management and linked to the write-only field. In these implementations, a change to the version field will prompt Pulumi to re-apply the write-only field's value to your cloud infrastructure.
```

Comment:
(minor) maybe add we should add a note like: "Notice that version fields are persisted into the state"